core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -186,8 +185,8 @@ object Uri extends UriFunctions {
 
 
   private def renderParamsAndFragment(writer: Writer, p: Query, f: Option[Fragment]): writer.type = {
-    if (p.nonEmpty) writer << '?' << p
-    if (f.isDefined) writer << '#' << UrlCodingUtils.urlEncode(f.get, spaceIsPlus = false)
+    if (!p.isEmpty) writer << p
```

Comment:
I'll be in and out the whole weekend so I should have time but I cant promise exactly when. Feel free to ping me whenever: if I'm not there I won't answer (at least right away...).